### PR TITLE
Feature: support telegram as a provider 

### LINKED ISSUE #18: Support Telegram as a provider for alerts
**Is your feature request related to a problem? Please describe.**
NA 

**Describe the solution you'd like**
Support Telegram as a provider.

### PR DESCRIPTION
**Pull request by Andrew44Ashraf**
Fixes #18  .

Changes proposed in this pull request:
- Support Telegram as a provider

### CODE REVIEW

File: provider/telegram.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	if err != nil || response.StatusCode > 202 {
+		return err
+	}
+
+	return nil
+}
+
+// SendMessage sends text message to the provider
+func (t *telegram) SendMessage(s string) error {
+	return nil
```
Comment (abahmed) [inline provider/telegram.go:80]: @Andrew44Ashraf Let's send normal message here

Comment (Andrew44Ashraf) [reply provider/telegram.go:104]: @abahmed updated please check

File: util/util.go [hunk 2]
```diff
@@ -109,6 +109,9 @@ func GetProviders() []provider.Provider {
 			if key == "discord" && c == "webhook" && len(strings.TrimSpace(v.(string))) > 0 {
 				providers = append(providers, provider.NewDiscord(viper.GetString("alert.discord.webhook")))
 			}
+			if key == "telegram" && c == "token" && len(strings.TrimSpace(v.(string))) > 0 {
+				providers = append(providers, provider.NewTelegram(viper.GetString("alert.telegram.token"), viper.GetString("alert.telegram.chatId")))
+			}
```
Comment (amgadramses) [inline util/util.go:114]: @Andrew44Ashraf do we need to add a check for the chat id as well?

File: provider/telegram.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+	logrus.Debugf("sending to telegram event: %v", e)
+
+	// validate telegram token and chat Id
+	err, _ := validateTelegram(t)
+	if err != nil {
+		return err
+	}
+
+	return sendByTelegramApi("", t, e)
+}
```
Comment (amgadramses) [inline provider/telegram.go:57]: @Andrew44Ashraf what do you think about moving the `buildRequestBodyTelegram` call to here, passing it as the first string parameter, and removing the event parameter from `sendByTelegramApi`

File: README.md [hunk 4]
```diff
@@ -95,6 +95,19 @@ If you want to enable PagerDuty, provide the integration key
...
+<p>
+    <img src="./assets/telegram.png" width="50%"/>
+</p>
+
+If you want to enable Telegram, provide a valid token and the chat Id.
+
+| Parameter                        | Description                                     |
+|:---------------------------------|:------------------------------------------------|
+| `alert.telegram.token`           | Telegram token                                  |
+| `alert.telegram.chatId`          | chat the you want to receive alert messages on  |
```
Comment (amgadramses) [inline README.md:109]: Typo in description

File: util/util.go (hunk 2 above)
Comment (Andrew44Ashraf) [reply util/util.go]: @abahmed updated

File: provider/telegram.go (hunk 3 above)
Comment (Andrew44Ashraf) [reply provider/telegram.go]: @amgadramses good one, refactored the code please check.

**Review by abahmed — APPROVED** [review]

**Pull request merged**